src/topology/homotopy/path.lean

```diff
@@ -257,6 +272,9 @@ lemma trans ⦃p₀ p₁ p₂ : path x₀ x₁⦄ (h₀ : p₀.homotopic p₁) (
 
 lemma equivalence : equivalence (@homotopic X _ x₀ x₁) := ⟨refl, symm, trans⟩
 
+lemma map {p q : path x₀ x₁} (h : p.homotopic q) (f : C(X, Y)) :
+  homotopic (p.map f.continuous) (q.map f.continuous) := ⟨h.some.map f⟩
```

Comment:
I think you can do instead
```suggestion
  homotopic (p.map f.continuous) (q.map f.continuous) := h.map (homotopy.map f)
```
Which uses `nonempty.map` and doesn't use `choice`

Reply:
Is this something I should use in general? So `nonempty.map` instead of `nonempty.some` whenever possible?

Reply:
It's not really a big deal; it depends where your threshold for "I'll just use the axiom of choice, it's easier" is. For me, "it saves a few characters" isn't past that threshold.

Instead of `map` you can also use `let ⟨h'⟩ := h in ⟨h'.map f⟩` which is also constructive.

Reply:
For everything in this folder, I haven't really given much thought about what would/wouldn't be constructive, since (I know it's not _the same thing_) everything is noncomputable anyways. But I think the code looks "nicer" with `map` so I'll switch to that.